pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -68,7 +68,8 @@ type stateData struct {
 	claims []*resourceapi.ResourceClaim
 
 	// Allocator handles claims with structured parameters.
-	allocator structured.Allocator
+	allocator      structured.Allocator
+	allocateClaims []*resourceapi.ResourceClaim
```

Comment:
Nit: can we call this `claimsToAllocate` as before?

Except for this it looks good. Removing `ClaimsToAllocate()` is justification enough for this change.